docs/fsdpv2.md

```diff
@@ -0,0 +1,58 @@
+# Fully Sharded Data Parallel via SPMD
+
+Fully Sharded Data Parallel via SPMD or FSDPv2 is an utility that re-epxresses the famous FSDP algorithm in SPMD. [This](https://github.com/pytorch/xla/blob/master/torch_xla/experimental/spmd_fully_sharded_data_parallel.py) is
+an experimental feature that aiming to offer a familiar interface for users to enjoy all the benefits that SPMD brings into
+the table. The design doc is [here](https://github.com/pytorch/xla/issues/6379).
+
+Example usage:
+```python3
+import torch
+import torch_xla.core.xla_model as xm
+import torch_xla.distributed.spmd as xs
+from torch_xla.experimental.spmd_fully_sharded_data_parallel import SpmdFullyShardedDataParallel as FSDPv2
+
+# Define the mesh following common SPMD practice
+num_devices = xr.global_runtime_device_count()
+mesh_shape = (num_devices, 1)
+device_ids = np.array(range(num_devices))
+# To be noted, the mesh must have an axis named 'fsdp', which the weights and activations will be sharded on.
+mesh = Mesh(device_ids, mesh_shape, ('fsdp', 'model'))
+
+# As normal FSDP, but an extra mesh is needed.
+model = FSDPv2(my_module, mesh)
+optim = torch.optim.Adam(model.parameters(), lr=0.0001)
+output = model(x, y)
```

Comment:
We'll need to add input sharding here - do you want to add an example with MpDeviceLoader?

Reply:
Let's manually shard the x here. Good catch!